### PR TITLE
MOTECH-2465 Button to toggle Tasks field bubbles on and off

### PR DESCRIPTION
**Pull request by wstrzelczyk**

### CODE REVIEW

File: modules/tasks/tasks/src/main/resources/webapp/partials/widgets/content-editable-map.html [hunk 1]
```diff
@@ -8,8 +8,8 @@
         </div>
     </div>
     <div class="clearfix">
-        <div contenteditable ng-model="pair.key" droppable class="actionField map-input map-field word-wrap input-extra-fluid" data-type="text"></div>
-        <div contenteditable ng-model="pair.value" droppable class="actionField map-input map-field word-wrap input-extra-fluid" data-type="text"></div>
+        <div ng-if="true" contenteditable ng-model="pair.key" droppable class="actionField map-input map-field word-wrap input-extra-fluid" data-type="text"></div>
```
Comment (sebbrudzinski) [inline modules/tasks/tasks/src/main/resources/webapp/partials/widgets/content-editable-map.html]: ?

File: modules/tasks/tasks/src/main/resources/webapp/partials/widgets/content-editable-map.html [hunk 2]
```diff
@@ -8,8 +8,8 @@
...
+        <div ng-if="true" contenteditable ng-model="pair.value" droppable class="actionField map-input map-field word-wrap input-extra-fluid" data-type="text"></div>
```
Comment (sebbrudzinski) [inline modules/tasks/tasks/src/main/resources/webapp/partials/widgets/content-editable-map.html]: ?

File: modules/tasks/tasks/src/main/resources/webapp/partials/widgets/content-editable-map.html [hunk 3]
```diff
@@ -19,8 +19,8 @@
     </div>
     <div class="clearfix" ng-repeat="pair in pairs">
         <div class="clearfix">
-            <div contenteditable ng-model="pair.key" droppable class="actionField map-input map-field word-wrap input-extra-fluid" ng-change="updateMap(pair, $index)" data-type="text"></div>
-            <div contenteditable ng-model="pair.value" droppable class="actionField map-input map-field word-wrap input-extra-fluid" ng-change="updateMap(pair, $index)" data-type="text"></div>
+            <div ng-if="true" contenteditable ng-model="pair.key" droppable class="actionField map-input map-field word-wrap input-extra-fluid" ng-change="updateMap(pair, $index)" data-type="text"></div>
```
Comment (sebbrudzinski) [inline modules/tasks/tasks/src/main/resources/webapp/partials/widgets/content-editable-map.html]: ?

File: modules/tasks/tasks/src/main/resources/webapp/partials/widgets/content-editable-map.html [hunk 4]
```diff
@@ -19,8 +19,8 @@
...
+            <div ng-if="true" contenteditable ng-model="pair.value" droppable class="actionField map-input map-field word-wrap input-extra-fluid" ng-change="updateMap(pair, $index)" data-type="text"></div>
```
Comment (sebbrudzinski) [inline modules/tasks/tasks/src/main/resources/webapp/partials/widgets/content-editable-map.html]: ?

File: modules/tasks/tasks/src/main/resources/webapp/js/directives.js [hunk 5]
```diff
@@ -458,15 +459,21 @@
...
+                if (!ngModel) {
                     return false;
                 }
 
                 scope.$watch(function () {
                     adjustText();
                     return ngModel.$viewValue;
-                }, function(){
-                    ngModel.$render();
+                }, function(value){
```
Comment (kmadej) [inline modules/tasks/tasks/src/main/resources/webapp/js/directives.js]: do we need this value variable ?

File: modules/tasks/tasks/src/main/resources/webapp/js/directives.js [hunk 6]
```diff
@@ -477,31 +484,80 @@
...
+                    scope.debug = args.debug;
+                    scope.changeBubble(ngModel, element);
+                });
+
+                scope.changeBubble = function (ngModel, element) {
+                    if (scope.data && scope.data.type !== 'MAP' && (scope.data.value === null || scope.data.value === "")) {
+                        return;
+                    }
+                    if (scope.data && scope.data.type === 'MAP') {
+                        if (scope.debug) {
```
Comment (kmadej) [inline modules/tasks/tasks/src/main/resources/webapp/js/directives.js]: this if can be done simplier:

File: modules/tasks/tasks/src/main/resources/webapp/js/directives.js [hunk 7]
```diff
@@ -477,31 +484,80 @@
...
+                                ngModel.$rollbackViewValue();
+                            }
+                            if (element.attr('ng-model') === "pair.key") {
+                                element.html(scope.pair.key);
+                                ngModel.$setViewValue(readContent(element));
+                                ngModel.$rollbackViewValue();
+                            }
+                        }
+                    } else {
+                        if (scope.debug) {
```
Comment (kmadej) [inline modules/tasks/tasks/src/main/resources/webapp/js/directives.js]: simplify this condition

File: modules/tasks/tasks/src/main/resources/webapp/messages/messages.properties [hunk 8]
```diff
@@ -255,6 +255,7 @@ task.examples=Examples
 task.pattern=Pattern
 task.formatt=Formatted Text
 task.thePatternSyntax=The pattern syntax is mostly compatible with java.text.SimpleDateFormat - time zone names cannot be parsed and a few more symbols are supported. All ASCII letters are reserved as pattern letters, which are defined as follows\:
+task.debug=Display raw syntax for fields
```
Comment (kmadej) [inline modules/tasks/tasks/src/main/resources/webapp/messages/messages.properties]: display raw syntax

Comment (kmadej) [inline modules/tasks/tasks/src/main/resources/webapp/messages/messages.properties]: change also message property name to task.bubbles.rawSyntax

**Pull request merged**